src/Sema.zig

```diff
@@ -6448,6 +6448,8 @@ fn analyzeCall(
 
     const callee_ty = sema.typeOf(func);
     const func_ty = func_ty: {
+        const msg = try sema.errMsg(block, func_src, "type '{}' is not a function", .{callee_ty.fmt(sema.mod)});
+        errdefer msg.destroy(sema.gpa);
```

Comment:
Analysis hasn't necessarily failed yet at this point so you shouldn't create an error message.